### DISCUSSION TITLE
Failure to run (secp256k1-sys)

### DISCUSSION THREAD

**Discussion by TomaszKot2** [Congyuwang/Rusty-Bitcoin-Explorer#7] [Q&A]
When I try to run the code i get following error:
error: failed to run custom build command for `secp256k1-sys v0.4.1`

Code:
use bitcoin_explorer::{BitcoinDB, FConnectedBlock, SConnectedBlock};
use std::path::Path;

fn main() {
    let path = Path::new("D:/BTC/Chain");
let db = BitcoinDB::new(path, false).unwrap();
}

Cargo.toml:
[package]
name = "BTC_explorer_rust"
version = "0.1.0"
edition = "2018"
[dependencies]
bitcoin-explorer = { git = "https://github.com/Congyuwang/Rusty-Bitcoin-Explorer" }

I have gcc also installed. System Win10 x64.

**Comment by Congyuwang** [comment]
Can you run `cmake` in your cmd window?

**Comment by Congyuwang** [comment]
I used to build this on Windows10 x86, using VS environment.

**Comment by TomaszKot2** [comment]
Yes, I can run cmake.

**Comment by Congyuwang** [comment]
Is there any more error message?

**Comment by TomaszKot2** [comment]
No, in rust only "failed to run custom build command for secp256k1-sys v0.4.1"

When running cmake on cmd, just standard information:
Usage
  cmake [options] <path-to-source>
  cmake [options] <path-to-existing-build>
  cmake [options] -S <path-to-source> -B <path-to-build>

Specify a source directory to (re-)generate a build system for it in the
current working directory.  Specify an existing build directory to
re-generate its build system.

Run 'cmake --help' for more information.

**Comment by Congyuwang** [comment]
I am not sure what your environment is like. So, I cannot really get an answer for you.

FYI, I updated the crate to v1.1.x, which almost guarantees to finish the deserialization (`db.iter_connected_block()`) within 1 hour (in your case, without considering the time for csv writing).

python is slow mainly because of the unsatisfactory performance of PyO3 crate.

**Comment by TomaszKot2** [comment]
Unfortunately I cannot make use of increased rust performance, because of this "secp256k1-sys v0.4.1" error. :/
Would it be possible for you to share some kind of executable rust file, giving the ability to export all input transactions (timestamp, adress, volume) to csv(s) and all output transactions to csv(s)?
I tried first in Python (on 1.0.5) but It would take around a week to parse the data.

**Reply by Congyuwang** [reply]
Is this csv exporter usable?

**Comment by Congyuwang** [comment]
I compiled a windows exe file here: https://github.com/Congyuwang/Blockchain_To_Csv/releases/tag/v0.1.0

The source code is in this repo:  https://github.com/Congyuwang/Blockchain_To_Csv, try modifying it to suit your need.

Just 100 lines of code using bitcoin-explorer crate.

**Comment by TomaszKot2** [comment]
Thank you very much for the exporter.
I started the run 3 days ago and it is still working (although on HDD not
SDD, maybe it is the cause?).
I think I will need to restart it on SDD.
Would it be possible for you to add an option to generate not one csv, but
to generate partial csvs (for example a new file after each "n" tdix
parsed)?
And maybe printing in the console the number of current block processed -
for example if it is divisible by 200 not to print too much data).
It would be very helpful, because this way it would be visible if it is
working correctly and what is the current progress.

Thank you very much for your support!

Best wishes,
Tomasz









pon., 11 paź 2021 o 06:07 Congyu ***@***.***> napisał(a):

**Reply by Congyuwang** [reply]
I think this should finish within hours.

Also it won't print too much data, because it only print a single line of progress bar, as shown in the picture here https://github.com/Congyuwang/Blockchain_To_Csv.

Yes, you may try to use SSD.

**Reply by Congyuwang** [reply]
You may try to modify the code to partition csv outputs. The code is here: https://github.com/Congyuwang/Blockchain_To_Csv.
If you cannot compile it on one computer, you may try on another PC to compile it, and copy the program back to where your data is located.
